src/code/z_bgcheck.c

```diff
@@ -1,432 +1,4713 @@
+            newNodeId = SSNodeList_GetNextNodeIdx(&colCtx->polyNodes);
+            SSNode_SetValue(&colCtx->polyNodes.tbl[newNodeId], &polyId, SS_NULL);
+            curNode->next = newNodeId;
+            return;
+        }
+
+        nextNode = &colCtx->polyNodes.tbl[curNode->next];
+        curPolyId = nextNode->polyId;
+
+        // if the poly being inserted is lower than the next poly
+        if (polyYMin < vtxList[COLPOLY_VTX_INDEX(polyList[curPolyId].flags_vIA)].y &&
+            polyYMin < vtxList[COLPOLY_VTX_INDEX(polyList[curPolyId].flags_vIB)].y &&
+            polyYMin < vtxList[polyList[curPolyId].vIC].y) {
+            newNodeId = SSNodeList_GetNextNodeIdx(&colCtx->polyNodes);
+            SSNode_SetValue(&colCtx->polyNodes.tbl[newNodeId], &polyId, curNode->next);
+            curNode->next = newNodeId;
+            return;
+        }
+        curNode = nextNode;
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/StaticLookup_AddPolyToSSList.s")
+#endif
+
+/**
+ * Add CollisionPoly to StaticLookup list
+ */
+void StaticLookup_AddPoly(StaticLookup* lookup, CollisionContext* colCtx, CollisionPoly* polyList, Vec3s* vtxList,
+                          s16 index) {
+    if (polyList[index].normal.y > COLPOLY_SNORMAL(0.5f)) {
+        StaticLookup_AddPolyToSSList(colCtx, &lookup->floor, polyList, vtxList, index);
+    } else if (polyList[index].normal.y < COLPOLY_SNORMAL(-0.8f)) {
+        StaticLookup_AddPolyToSSList(colCtx, &lookup->ceiling, polyList, vtxList, index);
+    } else {
+        StaticLookup_AddPolyToSSList(colCtx, &lookup->wall, polyList, vtxList, index);
+    }
+}
+
+/**
+ * Locates the closest static poly directly underneath `pos`, starting at list `ssList`
+ * returns yIntersect of the closest poly, or `yIntersectMin`
+ * stores the pointer of the closest poly to `outPoly`
+ * if (flags & 1), ignore polys with a normal.y < 0 (from vertical walls to ceilings)
+ */
+f32 BgCheck_RaycastFloorStaticList(CollisionContext* colCtx, u16 xpFlags, SSList* ssList, CollisionPoly** outPoly,
+                                   Vec3f* pos, f32 yIntersectMin, f32 chkDist, s32 flags, Actor* actor, s32 arg9) {
+    SSNode* curNode;
+    s32 polyId;
+    f32 result;
+    f32 yIntersect;
+    CollisionPoly* colPoly;
+    s32 pad;
+
+    result = yIntersectMin;
+    if (ssList->head == SS_NULL) {
+        return result;
+    }
+
+    curNode = &colCtx->polyNodes.tbl[ssList->head];
+
+    while (true) {
+        polyId = curNode->polyId;
+        colPoly = &colCtx->colHeader->polyList[polyId];
+
+        if (((flags & 1) && (colPoly->normal.y < 0)) ||
+            ((arg9 & 0x20) && ((colCtx->colHeader->surfaceTypeList[colPoly->type].data[0] >> 0x1E) & 1)) ||
+            COLPOLY_VIA_FLAG_TEST(colPoly->flags_vIA, xpFlags) ||
+            (COLPOLY_VIA_FLAG_TEST(colPoly->flags_vIB, 4) &&
+             ((actor != NULL && actor->category != ACTORCAT_PLAYER) || (actor == NULL && xpFlags != 1)))) {
```

Comment:
```suggestion
             ((actor != NULL && actor->category != ACTORCAT_PLAYER) || (actor == NULL && xpFlags != COLPOLY_IGNORE_CAMERA)))) {
```
You have these defines above for `xpFlags`, is this the right usage? There's a lot of `xpFlags != 1`, If this is the right use, this would be helpful in my camera docs to use these macros.